nemo/core/classes/common.py

```diff
@@ -369,21 +371,25 @@ def from_pretrained(
             A model instance of a particular model class
         """
         location_in_the_cloud = None
+        description = None
         if cls.list_available_models() is not None:
             for pretrained_model_info in cls.list_available_models():
                 if pretrained_model_info.pretrained_model_name == model_name:
                     location_in_the_cloud = pretrained_model_info.location
+                    description = pretrained_model_info.description
                     class_ = pretrained_model_info.class_
         if location_in_the_cloud is None:
             raise FileNotFoundError(
                 f"Model {model_name} was not found. Check cls.list_available_models() for the list of all available models."
             )
         filename = location_in_the_cloud.split("/")[-1]
         url = location_in_the_cloud.replace(filename, "")
-        cache_subfolder = f"NEMO_{nemo.__version__}"
+        cache_dir = Path.joinpath(Path.home(), f'.cache/torch/NeMo/NeMo_{nemo.__version__}/{filename[:-5]}')
```

Comment:
Why differentiate by NeMo Version?

Reply:
mostly to ease troubleshooting, but also to protect against (unlikely) scenario when only version changed, urls and descriptions stayed but checkpoints are incompatible.